src/cache_cf.cc

```diff
@@ -4726,7 +4723,8 @@ static void parse_ftp_epsv(acl_access **ftp_epsv)
         *ftp_epsv = nullptr;
 
         if (ftpEpsvDeprecatedAction == Acl::Answer(ACCESS_DENIED)) {
-            if (auto *a = Acl::Node::FindByName("all"))
+            static const auto all = new SBuf("all");
```

Comment:
What is the benefit of intentionally leaking a SBuf here versus
```
static const SBuf all("all");
```
?

This use case seems quite different from the NoNewGlobals one to me

Reply:
> What is the benefit of intentionally leaking a SBuf here versus

[ There is no real memory leak here, but I am ignoring that part of this question. ]

Same benefit as in NoNewGlobals: Unlike a [static local](https://en.cppreference.com/w/cpp/language/storage_duration#Static_local_variables) SBuf, the value of this POD variable is guaranteed to exist during destruction at program exit.

> This use case seems quite different from the NoNewGlobals one to me

In what way? The concerns here are essentially the same AFAICT. You can even imagine this code as the result of future/hypothetical `Acl::All().name` inlining.

Needless to say, we can spend a lot of time (in aggregate) debating whether specific functions might be called during at-exit destruction. One of the NoNewGlobals policy goals is to _avoid_ such debates in all contexts (at the expense of a few extra pointer dereferences in some contexts).

If you insist, we can use a local static SBuf here, but that will create a dangerous precedent by going against NoNewGlobals intent/goals.